pkg/doc/verifiable/credential.go

```diff
@@ -316,6 +378,128 @@ func NewCredential(dataJSON []byte, opts ...CredentialOpt) (*Credential, error)
 	return cred, nil
 }
 
+// credentialClaims is JWT Claims extension by Credential.
+type credentialClaims struct {
+	*jwt.Claims
+
+	Credential *rawCredential `json:"vc,omitempty"`
+}
+
+// rawCredentialClaims is used to get raw content of "vc" claim of JWT.
+type rawCredentialClaims struct {
+	*jwt.Claims
+
+	Raw map[string]interface{} `json:"vc,omitempty"`
+}
+
+func decodeJWT(rawJwt []byte, fetcher PublicKeyFetcher) ([]byte, *rawCredential, error) {
+	parsedJwt, err := jwt.ParseSigned(string(rawJwt))
+	if err != nil {
+		return nil, nil, fmt.Errorf("VC is not JWS: %w", err)
+	}
+
+	credClaims := new(credentialClaims)
+	err = parsedJwt.UnsafeClaimsWithoutVerification(credClaims)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to parse JWT claims: %w", err)
+	}
+
+	if verifyErr := verifyJWTSignature(parsedJwt, fetcher, credClaims); verifyErr != nil {
+		return nil, nil, verifyErr
+	}
+
+	// Apply VC-related claims from JWT.
+	credClaims.refineVCFromJWTClaims()
+
+	// Decode again to get raw content of "vc" claim.
+	rawClaims := new(rawCredentialClaims)
+	err = parsedJwt.UnsafeClaimsWithoutVerification(rawClaims)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to parse JWT claims: %w", err)
+	}
+
+	// Complement original "vc" JSON claim with data refined from JWT claims.
+	rawClaims.mergeRefinedVC(credClaims.Credential)
+
+	var vcData []byte
+	if vcData, err = json.Marshal(rawClaims.Raw); err != nil {
+		return nil, nil, errors.New("failed to marshal 'vc' claim of JWT")
+	}
+
+	return vcData, credClaims.Credential, nil
+}
+
+func (credClaims *credentialClaims) refineVCFromJWTClaims() {
+	raw := credClaims.Credential
+
+	if iss := credClaims.Issuer; iss != "" {
+		refineVCIssuerFromJWTClaims(raw, iss)
+	}
+
+	if nbf := credClaims.NotBefore; nbf != nil {
+		nbfTime := nbf.Time().UTC()
+		raw.Issued = &nbfTime
```

Comment:
is this intentional? `raw.Issued` instead of `raw.NotBefore`

Reply:
Yes, as `rawCredential.Issued` is written to both `iat` and `nbf` headers.